docs/blog/2019-11-19-gatsby-brings-speed-to-mediacurrent-com/index.md

```diff
@@ -0,0 +1,72 @@
+
+- A robust plugin system that easily integrates with Drupal and other systems
+- React and GraphQL-based
+- Highly optimized for performance
+
+Additionally, Gatsby’s streamlined development workflow allows us to make front-end design changes more quickly compared to working with a CSS-based templating system or Jekyll’s markdown files.
+
+## Migrating to Gatsby
+
+To replace Jekyll, we embarked on migrating our site’s front-end to Gatsby. We scoped the effort required which included converting Jekyll markdown files to Gatsby React components. The conversion process took less time than we estimated due to Gatsby’s clean, well-thought architecture. The back-end CMS configuration was minimal -- enable and configure Drupal’s JSON:API module. We had 7 content types on our site whose content would publish to Gatsby. Within each content type, we previously implemented Drupal’s Paragraphs module to allow structured content to be published to different devices. We found that in order to properly publish Paragraphs-based content, we needed to integrate it with GraphQL fragments.
+
+During the migration project, we ran into a series of unknowns that we hadn’t considered before starting development:
+
+- Build times
+  - The amount of content and assets were more than a usual Gatsby build. This caused our builds to time out in Netlify, our host.
+  - To resolve this, we started using a Jenkins build process which does the build, then pushes the build to Netlify. Since caching is still available on Jenkins, the build times have improved.
+  - More recently, we are working on leveraging [Gatsby Preview](https://www.gatsbyjs.com/preview/) which allows us to see changes _immediately_ on a preview site as well as the Gatsby Builds feature which cuts down our build times dramatically.
+- Image processing
+  - Processing images was one of the build processes that increased the build time. Consequently, many published images caused the build to be longer than expected. We worked on reducing and unpublishing older content and redirecting it to new, fresh content.
+- Body field processing.
+  - Images in the body field are not part of the regular processing and Drupal will render relative paths. Since these paths are not part of Gatsby, we needed to add logic that updates an image’s path to point to our Drupal instance URL.
+  - While we can access the images through the entity id, there was a second problem where the file would not transliterate the names and add the “%20” HTML character code where spaces are expected. At this time we are working on a patch for the gatsby-source-drupal, which changes the name of the file to replace %20 with a dash to make it more browser friendly. This will be contributed back to the community. In the meantime, we are redirecting any images inside a body field to our publishing system.
```

Comment:
```suggestion
  - While we can access the images through the entity id, there was a second problem where the file would not transliterate the names and add the `%20` HTML character code where spaces are expected. At this time we are working on a patch for the gatsby-source-drupal, which changes the name of the file to replace `%20` with a dash to make it more browser friendly. This will be contributed back to the community. In the meantime, we are redirecting any images inside a body field to our publishing system.
```